File: SudokuCollective.Domain/Models/Difficulty/Difficulty.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using SudokuCollective.Domain.Enums;
using SudokuCollective.Domain.Interfaces;

namespace SudokuCollective.Domain.Models {

    public class Difficulty: IDifficulty {

        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string DisplayName { get; set; }
        [Required]
        public DifficultyLevel DifficultyLevel { get; set; }
        public ICollection<SudokuMatrix> Matrices { get; set; }

        public Difficulty() {
            
            Id = 0;
            Name = string.Empty;
            DifficultyLevel = DifficultyLevel.NULL;
        }

        [JsonConstructor]
        public Difficulty(int id, string name, 
            string displayName, DifficultyLevel difficultyLevel) {
            
            Id = id;
            Name = name;
            DisplayName = displayName;
            DifficultyLevel = difficultyLevel;
        }
    }
}

File: SudokuCollective.Api/Client/src/models/difficulty.js
class Difficulty {
  constructor(data) {
    if (data == undefined) {
      this.id = 0;
      this.name = "";
      this.displayName = "";
      this.difficultyLevel = 0;
    } else {
      this.id = data.id;
      this.name = data.name;
      this.displayName = data.displayName;
      this.difficultyLevel = data.difficultyLevel;
    }
  }
}

export default Difficulty;
